### PR TITLE
Squeeze specification refactoring

### PR DESCRIPTION
**Pull request by blesniewski**
### Details:
- add a detailed description
- add a range specification for values of the 2nd input
- unify examples (add a short description to the 1st example)
- bold the required or optional keywoards in input descriptions

### Tickets:
 - *37483*
 - *45926*

### CODE REVIEW

File: docs/ops/shape/Squeeze_1.md [hunk 1]
```diff
@@ -4,29 +4,33 @@
...
 
 *   **1**: Tensor with squeezed values of type *T*.
 
 **Types**
 
-* *T*: supported type.
+* *T*: any supported type.
 
-* *T_SHAPE*: supported integer type.
+* *T_I*: any supported integer type.
```
Comment (pelszkow) [inline docs/ops/shape/Squeeze_1.md:29]: I use `T_INT` for such tensor. Not sure which approach is better.

File: docs/ops/shape/Squeeze_1.md [hunk 2]
```diff
@@ -4,29 +4,33 @@
...
 
 **Attributes**: *Squeeze* operation doesn't have attributes.
 
 **Inputs**:
 
-*   **1**: Multidimensional input tensor of type *T*. *Required*.
+*   **1**: Multidimensional input tensor of type *T*. **Required**.
 
-*   **2**: 0D or 1D tensor of type *T_SHAPE* with dimensions indices to squeeze. Values could be negative. *Optional*.
+*   **2**: 0D or 1D tensor of type *T_I* with indices of dimensions to squeeze. Values could be negative (have to be from range `[-R, R-1]`, where `R` is the rank of the first input). **Optional**.
```
Comment (pelszkow) [inline docs/ops/shape/Squeeze_1.md:19]: I'm not sure but I found `scalar` more frequently than `0D`.
Do you have any suggestion which is preferable.

File: docs/ops/shape/Squeeze_1.md (hunk 1 above)
Comment (blesniewski) [reply docs/ops/shape/Squeeze_1.md:29]: I suppose both are better than `T_SHAPE`, I will use you suggestion.

File: docs/ops/shape/Squeeze_1.md (hunk 2 above)
Comment (blesniewski) [reply docs/ops/shape/Squeeze_1.md:19]: I agree, will change it to `scalar`.

File: docs/ops/shape/Squeeze_1.md [hunk 3]
```diff
@@ -4,29 +4,33 @@
...
+*   **2**: Scalar or 1D tensor of type *T_INT* with indices of dimensions to squeeze. Values could be negative (have to be from range `[-R, R-1]`, where `R` is the rank of the first input). **Optional**.
 
 **Outputs**:
 
 *   **1**: Tensor with squeezed values of type *T*.
 
 **Types**
 
-* *T*: supported type.
+* *T*: any supported type.
```
Comment (jdanieck) [inline docs/ops/shape/Squeeze_1.md:27]: _T: any supported floating point or integer type._

**Review by jdanieck — APPROVED** [review]

File: docs/ops/shape/Squeeze_1.md (hunk 3 above)
Comment (blesniewski) [reply docs/ops/shape/Squeeze_1.md:27]: So, maybe numeric? I see it's pretty common.

**Review by pelszkow — APPROVED** [review]

File: docs/ops/shape/Squeeze_1.md [hunk 4]
```diff
@@ -4,29 +4,33 @@
 
 **Category**: Shape manipulation
 
-**Short description**: *Squeeze* removes specified dimensions (second input) equal to 1 of the first input tensor. If the second input is omitted then all dimensions equal to 1 are removed. If the specified dimension is not equal to one then error is raised.
+**Short description**: *Squeeze* removes dimensions equal to 1 from the first input tensor.
+
+**Detailed description**: *Squeeze* can be used with or without the second input tensor.
+*       If only the first input is provided, every dimension that is equal to 1 will be removed from it.
```
Comment (lazarevevgeny) [inline docs/ops/shape/Squeeze_1.md:10]: Why 8 spaces here?

**Review by lazarevevgeny — APPROVED** [review]

File: docs/ops/shape/Squeeze_1.md (hunk 4 above)
Comment (blesniewski) [reply docs/ops/shape/Squeeze_1.md]: I got confused about the spacing here. Changed it to one space now.

**Review by tomdol — APPROVED** [review]

**Pull request merged**